lib/logstash/inputs/wmi.rb

```diff
@@ -47,23 +47,28 @@ def run(queue)
     @wmi = WIN32OLE.connect("winmgmts://")
     
     begin
-      @logger.debug("Executing WMI query '#{@query}'")
-      loop do
+      while !stop?
+        @logger.debug("Executing WMI query '#{@query}'")
         @wmi.ExecQuery(@query).each do |wmiobj|
           # create a single event for all properties in the collection
           event = LogStash::Event.new
           event["host"] = @host
           decorate(event)
           wmiobj.Properties_.each do |prop|
-            event[prop.name] = prop.value
+            if prop.value.is_a?(String)
+              event[prop.name] = prop.value.force_encoding(Encoding::UTF_8)
```

Comment:
Is this right? It seems unrelated to the `stop` change. Plus Windows likes to use UTF-16 for most things, so, without testing, I'm not sure if this (UTF_8) Is correct.